services/yellow/src/api/Handlers/AuthorizeOrderCancellationFeeHandler.cs

```diff
@@ -22,40 +22,43 @@ public class AuthorizeOrderCancellationFeeHandler : AbstractPaymentHandler<Autho
             _paymentService = paymentService;
         }
         
-        protected override async Task<OrderCancellationFeeAuthorizationAcquired> Handle(AuthorizeOrderCancellationFeeRequest message)
+        public async Task Handle(AuthorizeOrderCancellationFeeRequest message, IMessageHandlerContext context)
         {
-            Order order = await Ensure(message, m => _orderService.FindOneById(m.OrderId));
+            try
+            {
+                var validateResult = ValidateRequest(message);
+                if (!validateResult.IsValid())
+                    throw new ValidationException(validateResult);
+                
+                var order = HandlerHelper.Ensure(await _orderService.FindOneById(message.OrderId))!;
             
-            var paymentMethods = order.PaymentMethods;
-            var paymentMethod = paymentMethods?.LastOrDefault();
-            var payToken = paymentMethod?.CreditCard.PaymentAuthorizationTokenId;
+                var paymentMethods = order.PaymentMethods;
+                var paymentMethod = paymentMethods?.LastOrDefault();
+                var payToken = paymentMethod?.CreditCard.PaymentAuthorizationTokenId;
             
-            if (payToken is null)
-                throw new InvalidOperationException($"The active payment method of Order {order.Id} is not suitable for payment: MissingToken");
+                if (payToken is null)
+                    throw new InvalidOperationException($"The active payment method of Order {order.Id} is not suitable for payment: MissingToken");
             
-            var authorizeResult = await _paymentService.Authorize(order.Price.CancellationFee, payToken!);
-            if (authorizeResult.IsSuccess != true)
-                throw authorizeResult.Exception ?? new InvalidOperationException($"{nameof(authorizeResult)}");
+                var authorizeResult = await _paymentService.Authorize(order.Price.CancellationFee, payToken!);
+                if (authorizeResult.IsSuccess != true)
+                    throw authorizeResult.Exception ?? new InvalidOperationException($"{nameof(authorizeResult)}");
             
-            return new OrderCancellationFeeAuthorizationAcquired(message.OrderId);
+                await context.Publish(new OrderCancellationFeeAuthorizationAcquired(message.OrderId));
+                await context.Reply(new Response<bool>(true));
+            }
+            catch (Exception e)
+            {
+                await context.Publish(new AuthorizeOrderCancellationFeeFailed(message.OrderId, e.Message));
+                await context.Reply(new Response<bool>(false));

```

Comment:
`Exception e` should be returned in the `Response`, instead of `false`

Reply:
Suggestion applied